src/label/sectorLabel.ts

```diff
@@ -0,0 +1,234 @@
+import {RectLike, Sector} from 'zrender';
+import {calculateTextPosition, TextPositionCalculationResult} from 'zrender/src/contain/text';
+import {BuiltinTextPosition, TextAlign, TextVerticalAlign} from 'zrender/src/core/types';
+import {isArray} from 'zrender/src/core/util';
+import {ElementCalculateTextPosition, ElementTextConfig} from 'zrender/src/Element';
+
+export type SectorTextPosition = BuiltinTextPosition
+    | 'startAngle' | 'insideStartAngle'
+    | 'endAngle' | 'insideEndAngle'
+    | 'middle'
+    | 'startArc' | 'insideStartArc'
+    | 'endArc' | 'insideEndArc'
+    | (number | string)[];
+
+export type SectorLike = {
+    cx: number
+    cy: number
+    r0: number
+    r: number
+    startAngle: number
+    endAngle: number
+    clockwise: boolean
+};
+
+export function createSectorCalculateTextPosition<T extends (string | (number | string)[])>(
+    positionMapping: (seriesLabelPosition: T) => SectorTextPosition
+): ElementCalculateTextPosition {
+    return function (
+        this: Sector,
+        out: TextPositionCalculationResult,
+        opts: {
+            position?: SectorTextPosition
+            distance?: number
+            global?: boolean
+        },
+        boundingRect: RectLike
+    ) {
+        const textPosition = opts.position;
+
+        if (!textPosition || textPosition instanceof Array) {
+            return calculateTextPosition(
+                out,
+                opts as ElementTextConfig,
+                boundingRect
+            );
+        }
+
+        const mappedSectorPosition = positionMapping(textPosition as T);
+        const distance = opts.distance != null ? opts.distance : 5;
+        const sector = this.shape;
+        const cx = sector.cx;
+        const cy = sector.cy;
+        const r = sector.r;
+        const r0 = sector.r0;
+        const middleR = (r + r0) / 2;
+        const startAngle = sector.clockwise ? sector.startAngle : sector.endAngle;
```

Comment:
I think we should always choose `sector.startAngle` here and put the clockwise logic in the mapping methods of `BarView`. Reasons:
1. The name of position is `startAngle` and `insideStartAngle`
2. Other charts may not have this `const clockwise = layout.startAngle < layout.endAngle` logic like `BarView` do. It's totally legal if they only use `clockwise` to control the direction. In this case, `startAngle` and `endAngle` should not be reversed.